lte/gateway/c/oai/tasks/s1ap/s1ap_mme_decoder.c

```diff
@@ -29,311 +33,28 @@
-
-    case S1ap_ProcedureCode_id_UEContextRelease: {
-      ret = s1ap_decode_s1ap_uecontextreleasecompleteies(
-          &message->msg.s1ap_UEContextReleaseCompleteIEs,
-          &successfullOutcome_p->value);
-      *message_id = S1AP_UE_CONTEXT_RELEASE_LOG;
-    } break;
-
-    case S1ap_ProcedureCode_id_UEContextModification: {
-      ret = s1ap_decode_s1ap_uecontextmodificationresponseies(
-          &message->msg.s1ap_UEContextModificationResponseIEs,
-          &successfullOutcome_p->value);
-      *message_id = S1AP_UE_CONTEXT_MODIFICATION_LOG;
-    } break;
-
-    case S1ap_ProcedureCode_id_E_RABSetup: {
-      ret = s1ap_decode_s1ap_e_rabsetupresponseies(
-          &message->msg.s1ap_E_RABSetupResponseIEs,
-          &successfullOutcome_p->value);
-    } break;
-
-    case S1ap_ProcedureCode_id_E_RABRelease: {
-      ret = s1ap_decode_s1ap_e_rabreleaseresponseies(
-          &message->msg.s1ap_E_RABReleaseResponseIEs,
-          &successfullOutcome_p->value);
-    } break;
-
-    default: {
-      OAILOG_ERROR(
-          LOG_S1AP,
-          "Unknown procedure ID (%ld) for successfull outcome message\n",
-          successfullOutcome_p->procedureCode);
-    } break;
-  }
-
-  return ret;
-}
-
-static int s1ap_mme_decode_unsuccessfull_outcome(
-    s1ap_message* message, S1ap_UnsuccessfulOutcome_t* unSuccessfulOutcome_p,
-    MessagesIds* message_id) {
-  int ret = RETURNerror;
-  DevAssert(unSuccessfulOutcome_p != NULL);
-  message->procedureCode = unSuccessfulOutcome_p->procedureCode;
-  message->criticality   = unSuccessfulOutcome_p->criticality;
-
-  switch (unSuccessfulOutcome_p->procedureCode) {
-    case S1ap_ProcedureCode_id_InitialContextSetup: {
-      ret = s1ap_decode_s1ap_initialcontextsetupfailureies(
-          &message->msg.s1ap_InitialContextSetupFailureIEs,
-          &unSuccessfulOutcome_p->value);
-      *message_id = S1AP_INITIAL_CONTEXT_SETUP_LOG;
-    } break;
-
-    case S1ap_ProcedureCode_id_UEContextModification: {
-      ret = s1ap_decode_s1ap_uecontextmodificationfailureies(
-          &message->msg.s1ap_UEContextModificationFailureIEs,
-          &unSuccessfulOutcome_p->value);
-      *message_id = S1AP_UE_CONTEXT_MODIFICATION_LOG;
-    } break;
-
-    default: {
-      OAILOG_ERROR(
-          LOG_S1AP,
-          "Unknown procedure ID (%d) for unsuccessfull outcome message\n",
-          (int) unSuccessfulOutcome_p->procedureCode);
-    } break;
-  }
-
-  return ret;
-}
-
-int s1ap_mme_decode_pdu(
-    s1ap_message* message, const_bstring const raw, MessagesIds* message_id) {
-  int ret = -1;
-
-  S1AP_PDU_t pdu         = {(S1AP_PDU_PR_NOTHING)};
-  S1AP_PDU_t* pdu_p      = &pdu;
-  asn_dec_rval_t dec_ret = {(RC_OK)};
-  DevAssert(raw != NULL);
-  memset((void*) pdu_p, 0, sizeof(S1AP_PDU_t));
+//-----------------------------------------------------------------------------
+int s1ap_mme_decode_pdu(S1ap_S1AP_PDU_t* pdu, const_bstring const raw) {
+  asn_dec_rval_t dec_ret;
+  DevAssert(pdu != NULL);
```

Comment:
In case of NULL pdu, we don't need to assert, we can return failed response along with log

Reply:
Resolved